modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -292,7 +284,7 @@ public void updateMessageID(String queueName, String nodeId, long startMessageId
             firstMessageId = startMessageIdInTheSlot;
         }
 
-	    if (slotRecoveryScheduled.compareAndSet(true, true)) {
```

Comment:
Using compareAndSet is redundant here since we just want to read (compare) the value here. i.e setting true when true is redundant. isn't it?